onig_sys/build.rs

```diff
@@ -64,73 +62,94 @@ fn link_type_override() -> Option<LinkType> {
+            #define HAVE_PROTOTYPES 1
+            #define STDC_HEADERS 1
+            #define HAVE_STRING_H 1
+            #define HAVE_STDARG_H 1
+            #define HAVE_STDLIB_H 1
+            #define HAVE_LIMITS_H 1
+            #define HAVE_INTTYPES_H 1
+            #define SIZEOF_INT 4
+            #define SIZEOF_SHORT 2
+            #define SIZEOF_LONG {}
+        ", if bits == "64" {"8"} else {"4"}
+        )).expect("Can't write config.h to OUT_DIR");
+    }
 
-#[cfg(target_env = "msvc")]
-fn compile(link_type: LinkType) {
-    let onig_sys_dir = env::current_dir().unwrap();
-    let build_dir = env::var("OUT_DIR").unwrap();
-    let lib_name = match link_type {
-        LinkType::Static => "onig_s",
-        LinkType::Dynamic => "onig",
-    };
-
-    let bitness = if cfg!(target_pointer_width = "64") {
-        "64"
-    } else {
-        "32"
-    };
-
-    // Execute the oniguruma NMAKE command for the chosen architecture.
-    let cmd = onig_sys_dir
-        .join("oniguruma")
-        .join(format!("make_win{}.bat", bitness))
-        .to_string_lossy()
-        .into_owned();
-    cmd!("cmd", "/c", cmd)
-        .dir(&build_dir)
-        .env_remove("MFLAGS")
-        .env_remove("MAKEFLAGS")
-        .read()
-        .unwrap();
-
-    println!("cargo:rustc-link-search=native={}", build_dir);
-    println!("cargo:rustc-link-lib={}={}", link_type, lib_name);
+    cc.include(out_dir); // Read config.h from there
+    cc.include(src);
+
+    let files = [
+        "regerror.c", "regparse.c", "regext.c", "regcomp.c",
+        "reggnu.c", "regenc.c", "regsyntax.c", "regtrav.c", "regversion.c",
+        "st.c", "regposerr.c", "onig_init.c",
+        "unicode.c", "ascii.c", "utf8.c", "utf16_be.c", "utf16_le.c",
+        "utf32_be.c", "utf32_le.c", "euc_jp.c", "sjis.c", "iso8859_1.c",
+        "iso8859_2.c", "iso8859_3.c", "iso8859_4.c", "iso8859_5.c",
+        "iso8859_6.c", "iso8859_7.c", "iso8859_8.c", "iso8859_9.c",
+        "iso8859_10.c", "iso8859_11.c", "iso8859_13.c", "iso8859_14.c",
+        "iso8859_15.c", "iso8859_16.c",
+        "euc_tw.c", "euc_kr.c", "big5.c", "gb18030.c", "koi8_r.c",
+        "cp1251.c", "euc_jp_prop.c", "sjis_prop.c",
+        "unicode_unfold_key.c", "unicode_fold1_key.c",
+        "unicode_fold2_key.c", "unicode_fold3_key.c",
+    ];
+    for file in files.iter() {
+        cc.file(src.join(file));
+    }
+
+    if cfg!(feature = "posix-api") {
+        cc.file(src.join("regexec.c"));
+    }
+
+    cc.warnings(false); // not actionable by the end user
+    cc.compile("onig");
 }
 
 pub fn main() {
-    if env_var_bool("RUSTONIG_SYSTEM_LIBONIG").unwrap_or(true) {
-        if let Ok(_) = Config::new().atleast_version("6.8.0").probe("oniguruma") {
+    let link_type = link_type_override().unwrap_or(DEFAULT_LINK_TYPE);
+    let use_pkg_config = env_var_bool("RUSTONIG_SYSTEM_LIBONIG")
+        .unwrap_or(link_type == LinkType::Dynamic); // pkg-config doesn't always support static
```

Comment:
Do we want to default to true here as the `statik` call in pkg config should cover the case when static isn’t supported?

Reply:
I don't trust pkg-config's `statik`. It tends to link dynamically even if you request static, which is why I'm skipping it.